packages/karma-esm/src/middleware.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+const path = require('path');
+const esm = require('./karma-esm');
+
+/**
+ * Middleware for serving es modules. Karma doesn't serve any files not specified
+ * beforehand. This is extremely inconvenient when working with es modules, as many
+ * files are requested on the fly.
```

Comment:
will this work for components that dynamically import from constructed urls?

like, imagine

```js
connectedCallback() {
  import(`https://cdn.components.thing/${this.someAttr}/component.js`)
    .then(this.doWithModule)
}
```

Reply:
Yes, because these requests will never reach the web server as the browser will request it from the internet.